src/utils/hooks/useCreateReview.tsx

```diff
@@ -1,45 +1,67 @@
-import { useState } from 'react'
-import { useCreateReviewMutation } from '../../utils/generated/graphql'
+import { Decision, ReviewInput, Trigger, useCreateReviewMutation } from '../generated/graphql'
 import { AssignmentDetails, FullStructure } from '../types'
+import { useGetFullReviewStructureAsync } from './useGetReviewStructureForSection'
 
-interface UseCreateReviewProps {
+// below lines are used to get return type of the function that is returned by useCreateReviewMutation
+type UseCreateReviewMutationReturnType = ReturnType<typeof useCreateReviewMutation>
+type PromiseReturnType = ReturnType<UseCreateReviewMutationReturnType[0]>
+// hook used to start a review, , as per type definition below (returns promise that resolve with mutation result data)
+type UseCreateReview = (props: {
   structure: FullStructure
   assignment: AssignmentDetails
-}
+}) => () => PromiseReturnType
+
+type ConstructReviewInput = (structure: FullStructure) => ReviewInput
 
-const useCreateReview = ({ assignment }: UseCreateReviewProps) => {
-  const [error, setError] = useState<ApolloError | undefined>()
+const useCreateReview: UseCreateReview = ({ structure, assignment }) => {
+  const [updateReview] = useCreateReviewMutation()
 
-  const [createReviewMutation] = useCreateReviewMutation({
-    onError: (error) => setError(error),
+  const getFullReviewStructureAsync = useGetFullReviewStructureAsync({
+    fullApplicationStructure: structure,
+    reviewAssignment: assignment,
   })
 
-  const createReviewFromStructure = async (structure: FullStructure) => {
+  const constructReviewInput: ConstructReviewInput = (structure) => {
     const elements = Object.values(structure?.elementsById || {})
-    // Exclude not assigned, not visible and missing responses
-    const reviewableElements = elements.filter(
-      (element) => element?.isAssigned && element?.element.isVisible && element.response?.id
+    // Only create review for elements that are pendingReview
+    const reviewableElements = elements.filter(({ isPendingReview }) => isPendingReview)
+
+    const reviewResponseCreate = reviewableElements.map(
+      ({ latestPreviousLevelReviewResponse, response, reviewQuestionAssignmentId }) => {
+        // link to applicaiton response or review response based on review level
+        const applicationResponseId = assignment.level > 1 ? undefined : response?.id
+        const reviewResponseLinkId =
+          assignment.level === 1 ? undefined : latestPreviousLevelReviewResponse?.id
+        return {
+          applicationResponseId,
+          reviewResponseLinkId,
+          reviewQuestionAssignmentId,
+        }
+      }
     )
 
-    const applicationResponses = reviewableElements.map((element) => ({
-      applicationResponseId: element.response?.id || 0,
-      reviewQuestionAssignmentId: element.assignmentId,
-    }))
-    const result = await createReviewMutation({
+    return {
+      trigger: Trigger.OnReviewCreate,
```

Comment:
We though it was better to set this on the mutation to make sure it is passed, would this way of doing this (using `ReviewInput`) have this as a compulsory input field?

Reply:
hmm, i see your point, was wondering what the difference is between setting it in mutation or here ? is it to protect someone using that mutation outside of this hook ?  (can we say only usable with this hook in the query comment, would that be sufficient ?

I would have to change to another type of mutation to set it explicitly in mutation, i was tying to avoid `ReviewResponseReviewIdFkeyReviewResponseCreateInput` etc.. which would need to be used

We can make it not null in the db schema, that would protect it from having a null value.

Reply:
Well, it needs to be set to make the action work. I guess it's fine to have it included as part of the hook, since we are the ones using it and we know this is required. A comment will be enough, but it can't be `null` since that's how we use to set a already processed trigger - changing it to null.

Reply:
ohh true, yeah it can't be null (the more we use nulls btw, the more I think we should avoid using nulls for logic : )